### PR TITLE
Showcase some possible bugs in Display and DebugCustom

### PR DESCRIPTION
**Pull request by JelteF**
I played around a bit with DebugCustom and Display after and found some
edge cases that are handled strangely. Are these bugs?

This is not meant to be merged as is. But more as an easy way to share some test
cases with (to me) unexpected results.

Related to #223

### CODE REVIEW

File: tests/display.rs [hunk 1]
```diff
@@ -100,6 +100,7 @@ mod structs {
             assert_eq!(Tuple(0).to_string(), "0");
             assert_eq!(format!("{:b}", Binary(10)), "1010");
             assert_eq!(Struct { field: 0 }.to_string(), "0");
+            assert_eq!(format!("{:?}", Struct { field: 0 }), "0");
```
Comment (JelteF) [inline tests/display.rs:103]: @tyranron Based on your comment here I would call this unexpected: https://github.com/JelteF/derive_more/issues/223#issuecomment-1316015457

It also makes sense in another way, since this is the newtype derive.

File: tests/display.rs [hunk 2]
```diff
@@ -172,6 +180,33 @@ mod structs {
...
+                assert_eq!(
+                    format!(
+                        "{:?}",
+                        Struct {
+                            field1: 0,
+                            field2: 3
+                        }
+                    ),
+                    "Struct: 0 3"
+                );
```
Comment (JelteF) [inline tests/display.rs:192]: This one makes sense. This is meant as a showcase of what DebugCustom is currently useful for.

File: tests/display.rs [hunk 3]
```diff
@@ -172,6 +180,33 @@ mod structs {
...
+                assert_eq!(
+                    format!(
+                        "{:?}",
+                        Struct2 {
+                            field1: 0,
+                            field2: 3
+                        }
+                    ),
+                    "Struct2"
+                );
```
Comment (JelteF) [inline tests/display.rs:209]: Should these cases error?

File: tests/display.rs (hunk 1 above)
Comment (tyranron) [reply tests/display.rs:103]: @JelteF @ilslv hmm... I'd say that the current `DebugCustom` implementation is not useful at all, in the contrast with `Display` and other `Display`-like derives. The problem is that it follows `Display`-like semantics, but in reality this is not what we expect. I'd propose to remake `DebugCustom` in a separate PR to follow its own `Debug` semantics inherited from `std`'s `#[derive(Debug)]` with steroids in form of:
- structural bounding of type parameters (as we do for `Display`-like macros);
- allowing to change how field is rendered granularly with `#[debug("fmt string", fmt, args)]` attribute;
- fields ignoring with `#[debug(ignore)]`.

Comment (JelteF) [reply tests/display.rs:103]: The behaviour you propose sounds good to me.

File: tests/display.rs (hunk 3 above)
Comment (JelteF) [reply tests/display.rs:209]:
```rust
                assert_eq!(
                    Struct2 {
                        field1: 0,
                        field2: 3
                    }.to_string(),
                    "Struct2"
                );
```
@ilslv @tyranron do we want to reconsider this behaviour?

Comment (tyranron) [reply tests/display.rs:209]: The [described and implemented design](https://github.com/JelteF/derive_more/pull/225#issuecomment-1319892317) misses default multi-field struct case.

@ilslv should we error in this case?

Comment (ilslv) [reply tests/display.rs:209]: @JelteF @tyranron yes, this definitely should produce compile-time error.

Comment (tyranron) [reply tests/display.rs:209]: @ilslv please, make a fix in a separate PR. Thanks!

Comment (ilslv) [reply tests/display.rs:209]: Fixed in https://github.com/JelteF/derive_more/pull/243

**Comment by tyranron** [comment]
Ooops, GitHub has triggered auto-close on a comment link =/

**Comment by JelteF** [comment]
All issues shown here are now fixed
